### ISSUE TITLE
CTGOV2 query for location returning incorrect number of studies

### ISSUE THREAD

**Issue by machado-t** [rfhb/ctrdata#41]
CTGOV2 query https://clinicaltrials.gov/search?locStr=Portugal&country=Portugal, which shows 3,203 studies on the website, only retrieves 7 studies. I tested similar queries for other locations and got similar results. 
On the other hand, i noticed the query without the locStr parameter https://clinicaltrials.gov/search?country=Portugal works correctly and retrieves the expected number of studies. 
I'm still on package version 1.17.1 - I'm sorry if this has been solved already. 
Thank you.

**Comment by rfhb** [comment]
"Location" in is a complex lookup system in the webinterface of `CTGOV2`. For example, it maps user entries to combinations of facility name, city, state, country. The translation of a search URL to an API call for this register may not readily work for all cases. An update will be provided to cover more use cases.

**Comment by rfhb** [comment]
Implemented with further user information. Closed with 3b9089237df1228224d779f52e29031612b9db38. 

```r
# example showing that an error will be raised when a user's search
# in CTGOV2 does not have have relevant details for an API query:

q <- "https://clinicaltrials.gov/search?locStr=Gustave%20Roussy,%20Rue%20Edouard%20Vaillant,%20Villejuif,%20France&lat=48.7955043&lng=2.3485255&distance=50"
ctrLoadQueryIntoDb(queryterm = q, verbose = F, only.count = TRUE)
# Error: Parameter 'locStr' provided, but no 'country', 'state' or 'city'; 
# please check in CTGOV; e.g., the name of a trial site should go into 
# Facility Name in the webinterface or parameter 'locn' in the search URL.

# series of examples showing correct translation:

q <- "https://clinicaltrials.gov/search?locStr=Portugal&country=Portugal"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 3203

q <- "https://clinicaltrials.gov/search?country=Portugal"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 3203

q <- "https://clinicaltrials.gov/search?locStr=Lissabon,%20Portugal&country=Portugal&state=Lisbon&city=Lisbon"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 497

q <- "https://clinicaltrials.gov/search?country=Portugal&state=Lisbon&city=Lisbon"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 497

q <- "https://clinicaltrials.gov/search?country=Portugal&city=Lisbon"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 497

q <- "https://clinicaltrials.gov/search?country=Portugal&state=Lisbon"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 21 noting there is no such state

q <- "https://clinicaltrials.gov/search?locStr=Bavaria,%20Germany&country=Germany&state=Bavaria"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 911 but CTGOV2 webinterface erroneously shows 25632 trials for all of the country

q <- "https://clinicaltrials.gov/search?locStr=Maine&country=United%20States&state=Maine"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 2273

q <- "https://clinicaltrials.gov/search?locStr=Paris,%20France&country=France&state=%C3%8Ele-de-France&city=Paris"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# only 104 returned by API but CTGOV2 webinterface shows 12647 trials 
# for the same parameters, indicating that state is not useful here with API

q <- "https://clinicaltrials.gov/search?locStr=Paris,%20France&country=France&city=Paris"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 12654

q <- "https://clinicaltrials.gov/search?locn=Charité Campus Virchow Klinikum"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 207

q <- "https://clinicaltrials.gov/search?locn=Gustave%20Roussy"
ctrLoadQueryIntoDb(queryterm = q, verbose = TRUE, only.count = TRUE)
# 1643
```

**Comment by machado-t** [comment]
Great work!
